satpy/writers/cf_writer.py

```diff
@@ -235,29 +235,32 @@ def assert_xy_unique(datas):
 
 
 def link_coords(datas):
-    """Link datasets and coordinates.
+    """Link dataarrays and coordinates.
 
-    If the `coordinates` attribute of a data array links to other datasets in the scene, for example
+    If the `coordinates` attribute of a data array links to other dataarrays in the scene, for example
     `coordinates='lon lat'`, add them as coordinates to the data array and drop that attribute. In the final call to
     `xr.Dataset.to_netcdf()` all coordinate relations will be resolved and the `coordinates` attributes be set
     automatically.
 
     """
-    for ds_name, dataset in datas.items():
-        coords = dataset.attrs.get('coordinates', [])
+    for da_name, data in datas.items():
+        coords = data.attrs.get('coordinates', [])
```

Comment:
Could this variable be renamed to `declared_coordinates` for example? Would make it easier to read.